hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1304,7 +1312,15 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 	// always reconcile the release image (facilitates rolling forward)
 	hcp.Spec.ReleaseImage = hcluster.Spec.Release.Image
 
-	hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
+	if hcluster.Spec.Configuration != nil {
+		hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
+		// for compatibility with previous versions of the CPO, the hcp configuration should be
+		// populated with individual fields *AND* the previous raw extension resources.
+		hcp.Spec.Configuration.Items = configurationFieldsToRawExtensions(hcluster.Spec.Configuration)
+	} else {
+		hcp.Spec.Configuration = nil
```

Comment:
This basically means that if hcp.Spec.Configuration is nil we set it to nil? That doesn't make a lot of sense to me

Reply:
No, it means that if the HostedCluster.Spec.Configuration == nil (has been cleared by the user), we also need to clear any previously set value on the HostedControlPlane.Spec.Configuration